aplib.net-demo/Assets/Scripts/Entities/HealthComponent.cs

```diff
@@ -80,6 +84,7 @@ public void Reset()
     /// </summary>
     private void Die()
     {
+        _entitySound.Die();
```

Comment:
Definitely don't do this. Some entities with Health don't need sounds. This is what the Death event is for. You can subscribe the sound to it from another script, for example the EntitySound script.
```suggestion
```